src/main/kotlin/com/greenbird/utilihive/stubs/sftp/SftpServerTestContext.kt

```diff
@@ -0,0 +1,323 @@
+         * Value 0 means choosing any available port automatically.
+         * @param port the port. Must be between 0 and 65535, where 0 means choose automatically.
+         * @throws IllegalArgumentException if the port is not between 0 and 65535.
+         * @throws IllegalStateException if the server cannot be restarted.
+         */
+        @Suppress("MagicNumber")
+        set(port) {
+            require(port in 0..65535) {
+                ("Port cannot be set to $port because only ports between 0 and 65535 are valid.")
+            }
+            verifyWithSftpServerIsNotFinished("set port")
+            restartServer(port)
+        }
+
+    companion object {
+        fun withSftpServer(block: SftpServerTestContext.() -> Unit) {
+            val server = SftpServerTestContext(createFileSystem())
+            server.start(0)
+            server.use(block)
+        }
+
+        private val RANDOM = Random()
+
+        @Throws(IOException::class)
+        private fun createFileSystem(): FileSystem {
+            return MemoryFileSystemBuilder.newLinux().build("SftpServerTestContext-" + RANDOM.nextInt())
+        }
+    }
+
+    override fun close() {
+        withSftpServerFinished = true
+        server.stop()
+    }
+
+    /**
+     * Register a username with its password. After registering a username
+     * it is only possible to connect to the server with one of the registered
+     * username/password pairs.
+     *
+     * If `addUser` is called multiple times with the same username but
+     * different passwords then the last password is effective.
+     * @param username the username.
+     * @param password the password for the specified username.
+     * @return the server itself.
+     */
+    fun addUser(username: String, password: String): SftpServerTestContext {
+        usernamesAndPasswords[username] = password
+        return this
+    }
+
+    private fun restartServer(port: Int) =
+        try {
+            server.stop()
+            start(port)
+        } catch (e: IOException) {
+            throw IllegalStateException("The SFTP server cannot be restarted.", e)
+        }
+
+    /**
+     * Puts a text file to the SFTP folder. The file is available by the specified path.
+     * @param path the path to the file
+     * @param content the file's content
+     * @param encoding the encoding of the file
+     * @throws IOException if the file cannot be written
+     */
+    @Throws(IOException::class)
+    fun putFile(
+        path: String,
+        content: String,
+        encoding: Charset
+    ) {
+        val contentAsBytes = content.toByteArray(encoding)
+        putFile(path, contentAsBytes)
+    }
+
+    /**
+     * Puts a file to the SFTP folder. The file is available by the specified path.
+     * @param path the path to the file
+     * @param content the file's content
+     * @throws IOException if the file cannot be written
+     */
+    @Throws(IOException::class)
+    fun putFile(
```

Comment:
I suggest we remove this overload. Its very easy to create a ByteArrayInputStream instead.
Overloads are messy and old school :)

Reply:
Removed including related tests